server/aws-lsp-codewhisperer/src/language-server/dependencyGraph/dependencyGraph.ts

```diff
@@ -116,6 +117,25 @@ export abstract class DependencyGraph {
         return files.reduce((a, f) => a.concat(f), [])
     }
 
+    /**
+     * @param rootPath root folder to look for .gitignore files
+     * @returns list of files without those that are git ignored
+     */
+    async filterOutGitIgnoredFiles(rootPath: string, files: string[]): Promise<string[]> {
+        // Pattern to find .gitignore files with either windows or unix path styles
+        const gitIgnorePattern = /.*[\/\\]\.gitignore$/
```

Comment:
although not a huge risk, this regex seems to be of polynomial complexity.

I didn't test this but would something like `(?:^|/)\.gitignore(?=$|/)` work?